data/lua/wml-tags.lua

```diff
@@ -688,11 +688,36 @@ function wml_actions.store_side(cfg)
 	end
 end
 
--- This is the port of the old [modify_ai] into lua. It is different from wesnoth.modify_ai in that it uses a standard side filter.
--- I don't know why these functions were made to behave differently, but this seems to be the more powerful and useful one according
--- to mattsc's comments
 function wml_actions.modify_ai(cfg)
-	wesnoth.modify_ai_wml(cfg)
+	local sides = utils.get_sides(cfg)
+	local component
+	if cfg.action == "add" or cfg.action == "change" then
+		local start = string.find(cfg.path, "[a-z]+%[[a-z0-9_*]+%]$")
```

Comment:
This does not work in all cases.  It does not match 'candidate_action' and it does not work with empty closing []. It should be
`string.find(cfg.path, '[a-z_]+%[[a-z0-9_*]*%]$')`
I think (note: two changes).